paymentsheet/src/test/java/com/stripe/android/paymentelement/embedded/DefaultEmbeddedSheetLauncherTest.kt

```diff
@@ -68,97 +75,100 @@ internal class DefaultEmbeddedSheetLauncherTest {
             customerState,
             selection,
         )
-        manageContractCallbackCaptor.value.onActivityResult(result)
-        assertThat(customerStateHolder.customer.value).isEqualTo(customerState)
-        assertThat(selectionHolder.selection.value).isEqualTo(selection)
+
+        val callback = it.manageRegisterCall.callback.asCallbackFor<ManageResult>()
+        callback.onActivityResult(result)
+
+        assertThat(it.customerStateHolder.customer.value).isEqualTo(customerState)
+        assertThat(it.selectionHolder.selection.value).isEqualTo(selection)
     }
 
     @Test
     fun `manageActivityLauncher callback does not update state on non-complete result`() = testScenario {
-        customerStateHolder.setCustomerState(PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
+        it.customerStateHolder.setCustomerState(PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
         val result = ManageResult.Cancelled(customerState = null)
-        manageContractCallbackCaptor.value.onActivityResult(result)
-        assertThat(customerStateHolder.customer.value).isEqualTo(PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
+        val callback = it.manageRegisterCall.callback.asCallbackFor<ManageResult>()
+
+        callback.onActivityResult(result)
+
+        assertThat(it.customerStateHolder.customer.value).isEqualTo(PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
+        assertThat(it.selectionHolder.selection.value).isEqualTo(null)
     }
 
     @Test
     fun `manageActivityLauncher callback updates state on non-complete result`() = testScenario {
-        val result = ManageResult.Cancelled(customerState = PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
-        manageContractCallbackCaptor.value.onActivityResult(result)
-        assertThat(customerStateHolder.customer.value).isEqualTo(PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
+        it.customerStateHolder.setCustomerState(PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
+        val result = ManageResult.Cancelled(customerState = null)
+
+        val callback = it.manageRegisterCall.callback.asCallbackFor<ManageResult>()
+        callback.onActivityResult(result)
+
+        assertThat(it.customerStateHolder.customer.value).isEqualTo(PaymentSheetFixtures.EMPTY_CUSTOMER_STATE)
+        assertThat(it.selectionHolder.selection.value).isEqualTo(null)
     }
 
     @Test
     fun `onDestroy unregisters launchers`() = testScenario {
-        lifecycleOwner.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
-        assertThat(formActivityLauncher.unregisterTurbine.awaitItem()).isEqualTo(Unit)
-        assertThat(manageActivityLauncher.unregisterTurbine.awaitItem()).isEqualTo(Unit)
+        it.lifecycleOwner.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+        val formUnregisterCall = awaitUnregisterCall()
+        val manageUnregisterCall = awaitUnregisterCall()
+
+        assertThat(formUnregisterCall).isNotNull()
```

Comment:
Do we have a way to verify this is in fact the form call, and not something else?